tests-legacy/Endpoints/AbstractEndpointAdminTest.php

```diff
@@ -54,6 +73,54 @@ protected function employeeLogin()
         $cookieName = 'PrestaShop-' . md5(_PS_VERSION_ . 'psAdmin');
         $_COOKIE[$cookieName] = $cipherTool->encrypt($cookieContent);
         Cache::store('isLoggedBack' . 1, true);
+
+        $this->symfonyLogIn();
+    }
+
+    /**
+     * Emulates a real employee logged to the Back Office.
+     */
+    protected function symfonyLogIn()
+    {
+        $loggedEmployeeData = new \stdClass();
+        $loggedEmployeeData->email = 'demo@prestashop.com';
+        $loggedEmployeeData->id = 1;
+        $loggedEmployeeData->passwd = '';
+        $loggedEmployeeMock = new LoggedEmployee($loggedEmployeeData);
+
+        $tokenMock = $this
+            ->getMockBuilder(AbstractToken::class)
+            ->setMethods([
+                'getUser',
+                'getRoles',
+                'isAuthenticated',
+            ])
+            ->getMockForAbstractClass();
+
+        $tokenMock->expects($this->any())
+            ->method('getUser')
+            ->willReturn($loggedEmployeeMock);
+
+        $tokenMock->expects($this->any())
+            ->method('getRoles')
+            ->willReturn([new Role(AuthorizationOptions::DEFAULT_EMPLOYEE_ROLE)]);
+
+        $tokenMock->expects($this->any())
+            ->method('isAuthenticated')
+            ->willReturn(true);
+
+        $tokenStorageMock = $this->getMockBuilder(TokenStorage::class)
+            ->setMethods([
+                'getToken',
+            ])
+            ->disableAutoload()
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $tokenStorageMock->method('getToken')
+            ->willReturn($tokenMock);
+
+        SymfonyContainer::getInstance()->set('security.token_storage', $tokenStorageMock);
```

Comment:
Alright I get it, you replace the token storage with a mock, and then replace it with the real storage after the test!
Then maybe you should rename the backup value, and add a comment in setUp/tearDown methods